src/main/scala/com/typesafe/sbt/packager/Stager.scala

```diff
@@ -30,7 +30,9 @@ object Stager {
     // TODO - Config file user-readable permissions....
     for {
       (from, to) <- copies
-      if from.canExecute
+      // Only set executable permission if it needs to be set. Note: calling to.setExecutable(true) if it's already
+      // executable is undesirable for a developer using inotify to watch this file as it will trigger events again
+      if from.canExecute && !to.canExecute
```

Comment:
What happens when `to` doesn't exist? Is there an existence check necessary?

Reply:
I did wonder that, and tested by removing the staged script. When running `stage` again, it correctly recreates and sets the execute permission. 

From the docs https://docs.oracle.com/javase/8/docs/api/java/io/File.html#canExecute-- 


> Returns:
> true if and only if the abstract pathname exists and the application is allowed to execute the file


So it seems it returns false if the pathname does not exist.

I did just notice this other comment on the `canExecute` docs though, which may be a slight concern? 

> On some platforms it may be possible to start the Java virtual machine with special privileges that allow it to execute files that are not marked executable.

I guess if this is the case, `canExecute` could return true, so the execute permission may never be set. I find it fairly unlikely / don't see why people would be running in this mode though. I'm not sure where to look for information on this?